src/test/components/ButtonWithPanel.test.js

```diff
@@ -70,6 +72,15 @@ describe('shared/ButtonWithPanel', function() {
       />
     );
 
-    expect(button).toMatchSnapshot();
+    fireEvent.click(getByValue('My Button'));
+    expect(container.firstChild).toMatchSnapshot();
+
+    //it closes the panel when Esc key is pressed
+    fireEvent.keyDown(container, {
+      key: 'Escape',
+      keyCode: 27,
+      which: 27,
+    });
```

Comment:
While you're here, do you think you could write another test to check that it closes the panel when we _click_ outside of the panel ? The logic is in `ArrowPanel` currently but here we're really testing both components together so we can add it here.